pkg/vsphere/extraconfig/decode.go

```diff
@@ -334,7 +331,7 @@ func decodeMap(src DataSource, dest reflect.Value, prefix string, depth recursio
 	mapkeys, err := src(prefix)
 	if mapkeys == "" || err != nil {
 		log.Debugf("No value found in data source for maps keys \"%s\"", prefix)
-		return this
+		return reflect.ValueOf(nil)
```

Comment:
This requires every reference to a map element, pointer or otherwise, to check for nil. I'd prefer to have an empty map returned. If you must absolutely have a nil pointer in the case of a pointer to a map, then check for map size explicitly.